### PR TITLE
Improve `stat.py` module with `Literal` values

### PR DESCRIPTION
**Pull request by sobolevn**

### CODE REVIEW

**Comment by Akuli** [comment]
I believe stubtest checks that the constants are right. Can you deliberately change one of these constants off by 1, and then back to whatever it is, to confirm that stubtest indeed checks them?

**Comment by sobolevn** [comment]
Let's see. This commit should be broken: https://github.com/python/typeshed/pull/6705/commits/9b91a35c7b7df8ba43b87593df5c0f1cc2b916e8

**Comment by sobolevn** [comment]
It fails: https://github.com/python/typeshed/runs/4637004036?check_suite_focus=true#step:6:8

Output:

```

Run python tests/stubtest_stdlib.py
error: stat.ST_MODE variable differs from runtime type Literal[0]
Stub: at line 18
Literal[9999]
Runtime:
0
```

File: stdlib/stat.pyi [hunk 1]
```diff
@@ -14,81 +15,84 @@ def S_ISPORT(mode: int) -> int: ...
...
+FILE_ATTRIBUTE_OFFLINE: Literal[4096]
+FILE_ATTRIBUTE_READONLY: Literal[1]
+FILE_ATTRIBUTE_REPARSE_POINT: Literal[1024]
+FILE_ATTRIBUTE_SPARSE_FILE: Literal[512]
+FILE_ATTRIBUTE_SYSTEM: Literal[4]
+FILE_ATTRIBUTE_TEMPORARY: Literal[256]
+FILE_ATTRIBUTE_VIRTUAL: Literal[65536]
 
 if sys.platform == "win32" and sys.version_info >= (3, 8):
     IO_REPARSE_TAG_SYMLINK: int
```
Comment (Akuli) [inline stdlib/stat.pyi:98]: Can you change these to `Literal`s too? You can find the right values by setting them all to `Literal[123]` and looking at stubtest errors.

Comment (sobolevn) [reply stdlib/stat.pyi:98]: Sorry, I don't have a windows machine 😞 
It is not clear what these are from their definition: https://github.com/python/cpython/blob/f4c03484da59049eb62a9bf7777b963e2267d187/Modules/_stat.c#L595-L606

Maybe in the next PR?

**Review by Akuli — APPROVED** [review]
Thanks!

**Pull request merged**